### DISCUSSION TITLE
Updating to Bookworm

### DISCUSSION THREAD

**Discussion by brianlay714** [mikebrady/shairport-sync#1763] [General]
Built the latest SPS version on bookworm OS using RPi hardware that was previously working fine on the old version.
SPS appears to start ok but when try to send audio to it, it stops.

Tried looking for logs but can't find them. 

Any suggestions on troubleshooting?

**Comment by mikebrady** [comment]
Thanks for the post. Please take a look at [this issue](https://github.com/mikebrady/shairport-sync/issues/1749). It relates to the Docker image, but it's really about changes in Bookworm and may be applicable in your case.

**Comment by brianlay714** [comment]
The discussion about the other Bookworm issue was helpful but I'm still struggling with getting my system running.
I started again with fresh Bookworm and SPS installs and was able to get SPS running without crashing.
A client can connect and stream but I don't have audio...just metadata.
I'm using a hifiberry DAC which I believe has the correct driver loaded.  sps-alsa-explore shows the hifiberry as a valid audio destination.
`pi@baldwin:~/sps-alsa-explore $ ./sps-alsa-explore
> Device Full Name:    "hw:sndrpihifiberry"
  Short Name:          "hw:0"
  This device seems suitable for use with Shairport Sync.
  The following rate and format would be chosen by Shairport Sync in "auto" mode:
     Rate              Format
     44100             S32_LE
`
the SPS backend config is 
`// Back End Settings

// These are parameters for the "alsa" audio back end.
// For this section to be operative, Shairport Sync must be built with the following configuration flag:
// --with-alsa
alsa =
{
output_device = "hw:0"; // the name of the alsa output device. Use "shairport-sync -h" to discover the names of ALSA hardware devices. Use "alsamixer" or "apl>
mixer_control_name = "PCM"; // the name of the mixer to use to adjust output volume. No default. If not specified, no mixer is used and volume in adjusted in >
//      mixer_control_index = 0; // the index of the mixer to use to adjust output volume. Default is 0. The mixer is fully identified by the combination of t>
//      mixer_device = "default"; // the mixer_device default is whatever the output_device is. Normally you wouldn't have to use this.

//      output_rate = "auto"; // can be "auto", 44100, 88200, 176400 or 352800, but the device must have the capability.
output_format = "auto"; // can be "auto", "U8", "S8", "S16", "S16_LE", "S16_BE", "S24", "S24_LE", "S24_BE", "S24_3LE", "S24_3BE", "S32", "S32_LE" or "S32_BE" >

`
Next step was to see if I could get a USB DAC working instead of the hifiberry.

Any other thoughts on next steps?

**Comment by brianlay714** [comment]
Apparently I was using the wrong driver for the hifiberry DAC
Working as expected now.
